### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by mzgoddard**
Reopening #14 since making sb1-converter public seems to have disconnected the origin repository from the previous fork.

### CODE REVIEW

File: src/coders/byte-stream.js [hunk 1]
```diff
@@ -35,6 +77,14 @@ class ByteStream {
         }
     }
 
+    /**
+     * Write a value to the stream with a BytePrimitive defining how to do so
```
Comment (kchadha) [inline src/coders/byte-stream.js]:
```suggestion
     * Write a value to the stream (with a BytePrimitive defining how to do so)
```

File: src/coders/byte-stream.js [hunk 2]
```diff
@@ -59,6 +117,14 @@ class ByteStream {
         return st;
     }
 
+    /**
+     * Write bytes from the given Uint8Array array to the stream and increment
+     * the position.
+     * @param {Uint8Array} bytes - Bytes to write to the stream.
+     * @param {number=} [start=0] - Where in bytes to start writing from.
+     * @param {nubmmer=} [end=bytes.length] - Where in bytes to stop writing.
```
Comment (kchadha) [inline src/coders/byte-stream.js]: Maybe add whether bytes[end] will be written to or not. It looks like not..

Also there is a typo: `nubmmer` => `number`

```suggestion
     * @param {number=} [end=bytes.length] - Where in bytes to stop writing, excluding position at bytes[end].
```

File: src/squeak/byte-take-iterator.js [hunk 3]
```diff
@@ -1,13 +1,35 @@
+/**
+ * Let the passed iterator only take bytes up to a certain position.
```
Comment (kchadha) [inline src/squeak/byte-take-iterator.js]: This is a weird way to start a comment describing a class. Maybe:

```suggestion
 * An iterator that only takes bytes up to a certain position.
```

Comment (kchadha) [reply src/squeak/byte-take-iterator.js]: Alternatively, maybe just remove this line completely. The comment below makes sense without it.

File: src/squeak/field-iterator.js [hunk 4]
```diff
@@ -7,7 +7,20 @@ import {
...
 
+/**
+ * Consume values for the byte stream with a iterator-like interface.
+ */
 class Consumer {
+    /**
+     * @param {object} options - Define the consumer.
+     * @param {function} [options.type=Value] - The {@link Field} type to
+     * create.
+     * @param {BytePrimitive} options.read - How to read the third Field
```
Comment (kchadha) [inline src/squeak/field-iterator.js:18]: Maybe explain why the *third* Field argument is the special one?

File: src/squeak/fields.js [hunk 5]
```diff
@@ -38,16 +75,49 @@ class Value extends Field {
     }
 }
 
+/**
+ * A header for a FieldObject representing its class and how many fields are in
+ * the object.
+ *
+ * The Fields following are header up to its size should be collected into a
```
Comment (kchadha) [inline src/squeak/fields.js]: This sentence is a little confusing...

File: src/squeak/ids.js [hunk 6]
```diff
@@ -1,50 +1,181 @@
...
+     * The following byte is the number of fields in this object. */
     WATCHER_SLIDER: 174,
+
+    /** A variable field header. The next byte is the version of this object.
+     * The following byte is the number of fields in this object. */
     LIST_WATCHER: 175
 };
 
+/**
+ * A inverted map of TYPES. Map ids numbers to their string names.
```
Comment (kchadha) [inline src/squeak/ids.js]:
```suggestion
 * A inverted map of TYPES. Map id numbers to their string names.
```

File: src/squeak/ids.js [hunk 7]
```diff
@@ -1,50 +1,181 @@
...
-    // STATIC_STRING to differentiate it from STRING in this codebase.
+
+    /** A variable field header. The next byte is the version of this object.
+     * The following byte is the number of fields in this object.
+     *
+     * In Scratch 2 this is called String. To reduce confusion in the set of
+     * types, this is called STATIC_STRING in this converter. */
     STATIC_STRING: 105,
+
+    /** A variable field header. The next byte is the version of this object.
```
Comment (kchadha) [inline src/squeak/ids.js]: Is this one and are the ones below done or are they TODOs?

**Review by kchadha — COMMENTED** [review]
Copied over review comments from #14.

File: src/squeak/ids.js (hunk 7 above)
Comment (mzgoddard) [reply src/squeak/ids.js]: They are done. The descriptions I'm writing in this file is only meant to be the raw data in the `.sb` file stored in Field instances. As this comment states, this header, appears as two bytes in the `.sb` file, a version byte, and a size byte. I copied the comment as the comment will appear in the produced documentation that way. Everything after `99` works that way.

Hmm. I could move this byte information into the FieldObjectHeader and use `{@link FieldObjectHeader}` to make this shorter but retain that information.

**Comment by mzgoddard** [comment]
- Squash commited suggestions
- Change further documentation based on conversation on items without suggestions

**Review by kchadha — APPROVED** [review]
LGTM!

**Comment by mzgoddard** [comment]
- Rebase onto develop

**Pull request merged**